### PR TITLE
JAX: Add section 6.2 & 6.3 Param Management and Param Initialization

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_builders-guide/init-param.md [hunk 1]
```diff
@@ -27,12 +27,22 @@ By default, Keras initializes weight matrices uniformly by drawing from a range
 TensorFlow provides a variety of initialization methods both in the root module and the `keras.initializers` module.
 :end_tab:
 
-```{.python .input  n=1}
+:begin_tab:`jax`
+By default, Flax initializes weights using `jax.nn.initializers.lecun_normal`,
+i.e by drawing samples from a truncated normal distribution centered on 0 with
```
Comment (astonzhang) [inline chapter_builders-guide/init-param.md:32]: i.e.,

File: chapter_builders-guide/init-param.md [hunk 2]
```diff
@@ -27,12 +27,22 @@ By default, Keras initializes weight matrices uniformly by drawing from a range
...
+the standard deviation set as the squared root of $1 / fan\_in$ where `fan_in`
```
Comment (astonzhang) [inline chapter_builders-guide/init-param.md:33]:
```
$1 / \text{fan}_{\text{in}}$
```

`fan_in` -> $ \text{fan}_{\text{in}}$

File: chapter_builders-guide/init-param.md [hunk 3]
```diff
@@ -223,7 +285,13 @@ Here we define a subclass of `Initializer` and implement the `__call__`
 function that return a desired tensor given the shape and data type.
 :end_tab:
 
-```{.python .input  n=14}
+:begin_tab:`jax`
+Jax initialization functions take as arguments the `PRNG Key`, `shape` and
```
Comment (astonzhang) [inline chapter_builders-guide/init-param.md:289]: PRNGKey

File: chapter_builders-guide/init-param.md [hunk 4]
```diff
@@ -223,7 +285,13 @@ Here we define a subclass of `Initializer` and implement the `__call__`
...
+`dtype`. Here we implement the function `my_init` that return a desired
```
Comment (astonzhang) [inline chapter_builders-guide/init-param.md:290]: returns

File: chapter_builders-guide/init-param.md [hunk 5]
```diff
@@ -270,24 +338,44 @@ net(X)
...
+print(params['params']['layers_0']['kernel'][:, :2])
+```
+
+:begin_tab:`mxnet, pytorch, tensorflow`
 Note that we always have the option
 of setting parameters directly.
+:end_tab:
+
+:begin_tab:`jax`
+When initializing parameters in JAX and Flax, the parameters dict returned
```
Comment (astonzhang) [inline chapter_builders-guide/init-param.md:358]: Use proper English in main text.
parameters dict -> the dictionary of parameters, or the parameter dictionary

File: chapter_builders-guide/parameters.md [hunk 6]
```diff
@@ -75,15 +75,42 @@ X = tf.random.uniform((2, 4))
...
 Each layer's parameters are conveniently
 located in its attribute.
+:end_tab:
+
+:begin_tab:`jax`
+Flax and JAX decouple the model and the parameters as you
+might have observed in the models defined previously.
+When a model is defined via the `Sequential` class,
+we first need to initialize the network to generate
+the parameters dict. In the parameters dict, we can access
```
Comment (astonzhang) [inline chapter_builders-guide/parameters.md:110]: parameters dict

File: chapter_builders-guide/parameters.md [hunk 7]
```diff
@@ -143,6 +181,13 @@ That's why we need to request the value explicitly.
 In addition to the value, each parameter also allows us to access the gradient. Because we have not invoked backpropagation for this network yet, it is in its initial state.
 :end_tab:
 
+:begin_tab:`jax`
+Unlike the other frameworks, JAX doesn't keep a track of the gradients over the
+neural network parameters, instead the params and the network are decoupled.
```
Comment (astonzhang) [inline chapter_builders-guide/parameters.md:186]: Spell properly in main text unless they are variable names surrounded with accents.
params -> parameters

File: chapter_builders-guide/parameters.md [hunk 8]
```diff
@@ -143,6 +181,13 @@ That's why we need to request the value explicitly.
...
+It allows the user to express their computation as a
+Python function, and use the grad() transformation for the same purpose.
```
Comment (astonzhang) [inline chapter_builders-guide/parameters.md:188]:
```
the `grad` transformation
```

**Review by astonzhang — APPROVED** [review]
LGTM except for requested changes (see comments)
